### ISSUE TITLE
No space left on device

### ISSUE THREAD

**Issue by ramnageena11** [ablab/quast#260]
Hi,
I have started the QUAST analysis with for a metagenome assembly with 
319486 contigs, total 217428693 bp, min 200 bp, max 110411 bp, avg 680 bp, N50 724 bp

Metaquast command:
metaquast -o metaquast_result_cntrl -m 500 out_S23-25_contigs/final.contigs.fa -1 cntl_clean_R1.fastq -2 cntl_clean_R2.fastq -t 20

with storage space 500 Gb.

**Error: No space left on device.**

Traceback (most recent call last):
  File "/home/majorram/anaconda3/envs/quaST/lib/python3.10/logging/__init__.py", line 1104, in emit
    self.flush()
  File "/home/majorram/anaconda3/envs/quaST/lib/python3.10/logging/__init__.py", line 1084, in flush
    self.stream.flush()
OSError: [Errno 28] No space left on device
Call stack:
  File "/home/majorram/anaconda3/envs/quaST/bin/metaquast", line 431, in <module>
    logger.exception(exc_value)
  File "/home/majorram/anaconda3/envs/quaST/opt/quast-5.2.0/quast_libs/log.py", line 219, in exception
    self._logger.exception(e)
Message: OSError(28, 'No space left on device')
Arguments: ()
ERROR! exception caught!

How much space do I need to run the analysis?

Thanks
rgds
Ram

**Comment by alexeigurevich** [comment]
Hi Ram,
It is hard to estimate the disk space required for the MetaQUAST run (it is too case-by-case specific). Still, you have a relatively small (220 Mbp) metagenome, so I wouldn't expect the tool to consume more than several gigabytes of the hard drive (unless something buggy is happening, but such things have never been reported so far). You can also use `--space-efficient` option to save some space ([see more detail on it in the manual](https://quast.sourceforge.net/docs/manual.html#space_eff))  

Could it be that your 500 GB drive is already almost full before running MetaQuast? You can check this with the `df -h` command. If this is not the case, please reopen the issue.

**Comment by zwets** [comment]
Not sure if this applies to MetaQUAST, but couldn't this also be space on `/tmp`?

**Comment by alexeigurevich** [comment]
This is not likely unless the output directory is in `/tmp/`. 
Quast (and MetaQuast, etc) doesn't write to `/tmp` itself, and I think all embedded third-party tools don't do this, too (though I am not 100% sure). The error occurred in `File "/home/majorram/anaconda3/envs/quaST/lib/python3.10/logging/init.py", line 1104, in emit`, so when writing some message to a log fil (`metaquast.log` or `quast.log`) which is created inside the Quast output directory (specified via `-o` or `./quast_results/results_<date_time>/` by default).

**Comment by ramnageena11** [comment]
Hi,
I have made space (1.2Tb) and ran QUAST it worked.
I think it may be the tmp/ during running the command.

Thanks
rgds
Ram

**Comment by alexeigurevich** [comment]
Hi all,

I have added it here for future reference. I overlooked that the reads were used in the running command (`-1 cntl_clean_R1.fastq -2 cntl_clean_R2.fastq`). In this case, indeed, Meta/QUAST generates huge alignment files in `/tmp`! See #123 for more details. 

We will try to optimize that. Using reads is not super common (and not super informative) in our tools, so we haven't prioritized this issue so far but it seems that it should be done rather sooner than later :)
